### PR TITLE
feat(rds): attach description to database secret

### PR DESCRIPTION
**Pull request by robertd**
## Commit Message
feat(rds): attach description to database secret

Attach a description to secret created as part of the database instance. 
This allows to identify secrets belonging to the different stacks in the AWS console.

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license

### CODE REVIEW

**Comment by robertd** [comment]
@nija-at I'd like to be able to name parent stack in the description, but only username is what's available at that level, right?

**Comment by robertd** [comment]
@nija-at Do you think this can make it into v1.25.0 release? #6343

**Comment by nija-at** [comment]
> @nija-at I'd like to be able to name parent stack in the description, but only username is what's available at that level, right?

You can get the parent stack name using the method - [`Aws.STACK_NAME`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Aws.html#static-stack-name)

File: packages/@aws-cdk/aws-rds/lib/database-secret.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export class DatabaseSecret extends secretsmanager.Secret {
   constructor(scope: Construct, id: string, props: DatabaseSecretProps) {
     super(scope, id, {
       encryptionKey: props.encryptionKey,
+      description: `Auto-generated RDS secret for user: ${props.username}`,
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/lib/database-secret.ts]: Change this to something like: 'Generated by the CDK for stack: ${stackname}'

Comment (robertd) [reply packages/@aws-cdk/aws-rds/lib/database-secret.ts]: Will do.

**Comment by robertd** [comment]
Thanks for the tip @nija-at. I'll use the same verbiage when addressing #6235.

**Pull request merged**